Uplift/Views/DropdownView.swift

```diff
@@ -0,0 +1,201 @@
+//
+//  DropdownView.swift
+//  
+//
+//  Created by Cameron Hamidi on 10/7/19.
+//
+
+import Foundation
+import UIKit
+import SnapKit
+
+enum DropdownStatus {
+    case closed, half, open
+}
+
+enum DropdownDirection {
+    case horizontal, vertical
+}
+
+protocol DropdownViewDelegate: class {
+    func dropdownViewClosed(sender dropdownView: DropdownView)
+    func dropdownViewOpen(sender dropdownView: DropdownView)
+    func dropdownViewHalf(sender dropdownView: DropdownView)
+}
+
+class DropdownView: UIView {
+
+    // MARK: - INITIALIZATION
+    private var contentView: UIView!
+    private var contentViewHeight: CGFloat = 0
+    private weak var delegate: DropdownViewDelegate?
+    private var halfCloseDropdownGesture: UITapGestureRecognizer!
+    private var halfCloseView: UIView?
+    private var halfCloseViewHeight: CGFloat = 0
+    private var halfDropdownEnabled: Bool = false
+    private var halfHeight: CGFloat = 0
+    private var halfOpenDropdownGesture: UITapGestureRecognizer!
+    private var halfOpenView: UIView?
+    private var halfOpenViewHeight: CGFloat = 0
+    private var headerView: UIView!
+    private var headerViewHeight: CGFloat = 0
+    private var openCloseDropdownGesture: UITapGestureRecognizer!
+
+    var contentViewHeightConstraint: Constraint!
+    var currentHeight: CGFloat!
+    var status: DropdownStatus = .closed
+
+    init(headerView: UIView,
+         headerViewHeight: CGFloat,
+         contentView: UIView,
+         contentViewHeight: CGFloat,
+         halfDropdownEnabled: Bool = false,
+         halfOpenView: UIView? = nil,
+         halfOpenViewHeight: CGFloat = 0,
+         halfCloseView: UIView? = nil,
+         halfCloseViewHeight: CGFloat = 0,
+         halfHeight: CGFloat = 0) {
+        super.init(frame: .zero)
+
+        self.headerView = headerView
+        self.headerView.isUserInteractionEnabled = true
+        self.headerView.translatesAutoresizingMaskIntoConstraints = false
```

Comment:
if we're using Snapkit, I would think we shouldn't need to touch the translatesAutoresizingMaskIntoConstraints properties